source/mod_boinc.f90

```diff
@@ -141,37 +162,185 @@ subroutine boinc_post
 end subroutine boinc_post
 
 ! ================================================================================================ !
-!  Bump progress to 100%, and close the log file
+!  Sets the pre tracking progress.
+!  V.K. Berglyd Olsen, BE-ABP-HSS
+!  Created: 2019-05-21
+!  Updated: 2019-05-21
+!  These steps are hardcoded, and should only be changed if a time consuming pre tracking routine
+!  is added in main_cr. If so, bump the mSteps parameter and recheck all calls to this routine.
+!  Currently there are 5 steps:
+!   - Step 0 : Implicit, boinc_initialisation
+!   - Step 1 : Called after checpoint/restart init, called from main_cr
+!   - Step 2 : Called before daten, called from main_cr
+!   - Step 3 : Called after daten, called from main_cr
+!   - Step 4 : Called before first closed orbit call, called from main_cr
+!   - Step 5 : Called after linopt, called from main_cr
+!   - Step 6 : Called after closed orbit, called from main_cr
+!   - Step 7 : Called after beam distribution, called from main_cr
+!   - Step 8 : Called after pre-tracking, called from main_cr
+! ================================================================================================ !
+subroutine boinc_preProgress(nStep)
+  integer, intent(in) :: nStep
+  double precision :: progFrac
+  integer, parameter :: mSteps = 8
+  if(nStep <= mSteps) then
+    progFrac = dble(nStep)/dble(mSteps)/100.0
+  else
+    progFrac = 0.01
+  end if
+  call boinc_fraction_done(progFrac)
+  call boinc_summary(0)
+  write(boinc_logBuffer,"(a,f8.3,a)") "Progress: ",100.0*progFrac," %"
+  call boinc_writeLog
+end subroutine boinc_preProgress
+
+! ================================================================================================ !
+!  Sets the post tracking progress.
+!  V.K. Berglyd Olsen, BE-ABP-HSS
+!  Created: 2019-05-21
+!  Updated: 2019-05-21
+!  These steps are hardcoded, and should only be changed if a time consuming post processing routine
+!  is added in main_cr. If so, bump the mSteps parameter and recheck all calls to this routine.
+!  Currently there are 5 steps:
+!   - Step 0 : Called after tracking, called from main_cr
+!   - Step 1 : Called after reports, fort.12 and final state file, called from main_cr
```

Comment:
I don't see the Step 1 (at least in the diff)

Reply:
You're right. It was missing. Fixed.